.NET/Microsoft.Recognizers.Text.Number/Extractors/BaseNumberRangeExtractor.cs

```diff
@@ -0,0 +1,223 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Recognizers.Text.Number
+{
+    public abstract class BaseNumberRangeExtractor : IExtractor
+    {
+        private readonly BaseNumberExtractor numberExtractor;
+
+        private readonly BaseNumberExtractor ordinalExtractor;
+
+        internal abstract System.Collections.Immutable.ImmutableDictionary<Regex, string> Regexes { get; }
+
+        protected virtual string ExtractType { get; } = "";
+
+        public BaseNumberRangeExtractor(BaseNumberExtractor numberExtractor, BaseNumberExtractor ordinalExtractor)
+        {
+            this.numberExtractor = numberExtractor;
+            this.ordinalExtractor = ordinalExtractor;
+        }
+
+        public virtual List<ExtractResult> Extract(string source)
+        {
+            if (string.IsNullOrEmpty(source))
+            {
+                return new List<ExtractResult>();
+            }
+
+            var result = new List<ExtractResult>();
+            var matchSource = new Dictionary<Tuple<int, int>, string>();
+            var matched = new bool[source.Length];
```

Comment:
The default is not false?

Reply:
The default is false. I removed the redundant code.